ruby_on_rails/advanced_forms_and_activerecord/installing_psql.md

```diff
@@ -0,0 +1,342 @@
+### Introduction
+
+One of the great things about Rails is that it comes with some sensible defaults so that you can get started quickly. However, SQLite is typically not used for production. One major issue is that many hosting providers have ephemeral file systems which means your database can be deleted.
+
+For the apps written so far, it's probably not a big deal. As we progress to writing larger applications, though, it will become increasingly sensible to make sure that data persists even after a restart or hibernation.
+
+It is also sensible to make sure the application development environment is as close to the production environment as possible. This helps to avoid problems that stem from using different tools in different environments.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+-  What an environment is and how it can impact our application
+-  What environment variables are and how to utilize them
+-  How to install PostgreSQL and set it up
+-  How to set up a bare Rails application to use a specific database
+
+### Environments and Pipelines
+
+In the context of application development, an environment is the collection of tools, configurations and conventions that developers write and run code with. An environment usually consists of an operating system and other software packages on top of that operating system. Some of that software, like `code`, helps write software while others, like `rbenv`, provide tools and configurations to help run software.
+
+There are four basic types of environments that are typically set up together. The application that's being developed will then progress through these steps, which are grouped together in a "production pipeline":
+
+1,  The development environment is where developers can make changes, add features and improve the code without impacting actual users.
+1,  The test environment is set up to facilitate testing of the application. This environment may use different tools and configuration settings (like more verbose logging) to validate a program's behavior and identify bugs.
+1,  The staging environment usually mirrors the production environment but deployments are not meant to be public.
+1,  The production environment runs the application so that it is accessible by users.
```

Comment:
Fixing the commas!

```suggestion
1.  The development environment is where developers can make changes, add features and improve the code without impacting actual users.
1.  The test environment is set up to facilitate testing of the application. This environment may use different tools and configuration settings (like more verbose logging) to validate a program's behavior and identify bugs.
1.  The staging environment usually mirrors the production environment but deployments are not meant to be public.
1.  The production environment runs the application so that it is accessible by users.
```